src/hooks/useProviderSetup.ts

```diff
@@ -1,45 +0,0 @@
-import { useEffect, useState } from 'react';
-import { getFlashbotsProvider, getProvider } from '@/handlers/web3';
-import { StaticJsonRpcProvider } from '@ethersproject/providers';
-import { ethereumUtils } from '@/utils';
-import { getOnchainAssetBalance } from '@/handlers/assets';
-import { ParsedAddressAsset } from '@/entities';
-import { ChainId } from '@/chains/types';
-
-export const useProviderSetup = (chainId: ChainId, address: string) => {
-  const [provider, setProvider] = useState<StaticJsonRpcProvider | null>(null);
-  const [nativeAsset, setNativeAsset] = useState<ParsedAddressAsset | null>(null);
-
-  useEffect(() => {
-    const initProvider = async () => {
-      let p;
-      if (chainId === ChainId.mainnet) {
-        p = await getFlashbotsProvider();
```

Comment:
this hook was only used in the `SignTransactionSheet` for WalletConnect and should not be needing to use the flashbots provider for this flow; moved the `fetchNativeAsset` logic (lines 27) into `SignTransactionSheet` itself